{
  "affected": [
    {
      "database_specific": {
        "source": "https://github.com/bitnami/vulndb/tree/main/data/pillow/BIT-pillow-2023-44271.json"
      },
      "package": {
        "ecosystem": "Bitnami",
        "name": "pillow",
        "purl": "pkg:bitnami/pillow"
      },
      "ranges": [
        {
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "10.0.0"
            }
          ],
          "type": "SEMVER"
        }
      ],
      "severity": [
        {
          "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
          "type": "CVSS_V3"
        }
      ]
    }
  ],
  "aliases": [
    "CVE-2023-44271",
    "GHSA-8ghj-p4vj-mr35",
    "PYSEC-2023-227"
  ],
  "database_specific": {
    "cpes": [
      "cpe:2.3:a:python:pillow:*:*:*:*:*:*:*:*"
    ],
    "severity": "High"
  },
  "details": "An issue was discovered in Pillow before 10.0.0. It is a Denial of Service that uncontrollably allocates memory to process a given task, potentially causing a service to crash by having it run out of memory. This occurs for truetype in ImageFont when textlength in an ImageDraw instance operates on a long text argument.",
  "id": "BIT-pillow-2023-44271",
  "modified": "2025-04-03T14:40:37.652Z",
  "published": "2024-03-06T11:01:40.663Z",
  "references": [
    {
      "type": "WEB",
      "url": "https://devhub.checkmarx.com/cve-details/CVE-2023-44271/"
    },
    {
      "type": "WEB",
      "url": "https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7"
    },
    {
      "type": "WEB",
      "url": "https://github.com/python-pillow/Pillow/pull/7244"
    },
    {
      "type": "WEB",
      "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4/"
    },
    {
      "type": "WEB",
      "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html"
    },
    {
      "type": "WEB",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-44271"
    }
  ],
  "schema_version": "1.6.0"
}